### ISSUE TITLE
Mylar merging comics released in the same year.

### ISSUE THREAD

**Issue by leaderdog** [evilhero/mylar#2322]
Since I ran Maintenance on Mylar.db I've been running into weird issues.

This one is somewhat troubling.

Maintenance Merged these two Comics together:

Red Sonja (2016) ` https://comicvine.gamespot.com/red-sonja/4050-87337/`  V3
Red Sonja (2016) `https://comicvine.gamespot.com/red-sonja/4050-96671/` V4

I get that comic vine isn't terribly reliable on their naming of comics.  such as adding one-shot to the title of books, or putting the Company or author's names in front of the title and since some of these I believe were changed periodically by comicvine, it really makes a nightmare to fix because the maintenance went by the new name from comicvine.

However, merging two separate books that happen to be released in the same year is an obvious problem.

Is this just something we have to deal with because of comicvine? or is there something squirrely going on with the maintenance that it wasn't aware that these were two separate titles?  They should have had different comicvine ids so instead of using a v3 or v4 it just named it Red Sonja and stuffed all files in, overwriting or deleting some.

So I suppose the issue here is, When I rename these so they are in Separate folders again, if I ever have to do maintenance again, it would potentially do the same thing and create hours and hours of work to fix simply because of a naming issue in either comicvine or the way maintenance handled the  comic ids.

This one I happened to have caught because some were named V3 and V4 with one duplicate sitting named differently.  Which makes me wonder how many other titles are messed up like this.

**Comment by leaderdog** [comment]
It happened with Jupiter's Circle as well.
v2 `https://comicvine.gamespot.com/jupiters-circle/4050-81244/`
v3 `http://172.16.1.89:8090/comicDetails?ComicID=86239`

4 issues lost

**Comment by leaderdog** [comment]
It happened with Deadpool & The Mercs For Money
V1  `https://comicvine.gamespot.com/deadpool-the-mercs-for-money/4050-87793/
V2 `https://comicvine.gamespot.com/deadpool-the-mercs-for-money/4050-92360/`

Both are listed in Mylar.  And it randomly puts the issues in.  #4 in V1 was from V2.  The other folder that was originally labeled V2 didn't get affected.  And both coimic IDs in Mylar are corrected.

**Comment by leaderdog** [comment]
I don't know if you need any more info or not, but any comic that has the same name and came out in the same year is merged together during maintenance.
Blade Bunny (2016)
Guardians of the Galaxy (2015)

I'm going to just leave it at that, and wait for a response, as I'm curious as to why this happens beyond the fact they have the same name but different comic IDs.

**Comment by evilhero** [comment]
In your folder format string within the mylar configuration - do you have $VolumeN or $VolumeY or any volume designation at all.

This btw, is a combo problem  of CV naming and not distinguishing your folders via folder format to the right extent.

**Comment by leaderdog** [comment]
Awww, ok this I do not have.  I just have:
$Publisher\$Series ($Year)

If I add it now two questions:
1) does it show v1 or does it only start if there is a second volume?
2) would I just select all the books and hit rename?  or how would I go about getting it changed over now?

**Comment by evilhero** [comment]
Yes maintenance is meant to readd your existing watchlist back into your new db - it only knows the comicid and the series title. 

It has no knowledge of the location, so it has to recreate the location based off of your existing settings - ideally they should match up to what they were previously, provided that CV hasnt changed stuff or you have manually altered the path to something else. Maintenance also doesn't move any files, it updates the location in your dB and creates the directories if you have that option enabled.

Unless its a one-shot, it will tag it as a v1 in your directory structure.

And you can't rename folders within mylar - renaming files will have no impact on the folders. You could refresh the series after you updated the settings, and it would recreate the directories using those new settings. I don't believe it renames existing folder names to the new names either - it creates new, so the old directory would still exist and you'd have to manually move them to the new location.

**Comment by leaderdog** [comment]
oooh WoW  that would be a lot of work. :)
3700+ titles.

I guess I'd be better off making a file in my collection with the folders that duplicate names, and just repair those if I ever have to do maintenance again.

ok, thanks for clearing this all up.  :)

**Comment by evilhero** [comment]
Not all 3700+ titles are duplicates tho - so it's be probably less than 50 that have the same name and same year.

Next time you were to run maintenance mode (hopefully you don't, but if you need to), with your folder format containing the $VolumeN it will recreate directories so they won't match up to your existing structure either (since your existing structure doesn't contain $VolumeN).

**Comment by leaderdog** [comment]
yeah, 

that's sort of why I decided it wouldn't be a good idea to add the volume number except for the conflicting.  It's too much to manually change them all currently, and in case of a maintenance run, most would be not in their correct folders.  So I'll just leave it be.  Copying the few files over so I don't have to research for them later will work pretty smoothly. :)

I was thinking of doing an import.  But I've never done that before.  so not sure how that would work.  If it moves files from location to the next, then I could push my entire collection into a new folder (it's Currently just on the H:\ drive)  so say create a folder H:\Import\  and all the folders in there.  If it moves them all back to the H:\ drive then it would get the right volume number (I believe) plus I'm renaming everything with the day/month for each file.  It should be able to find the correct volume for everything.   But not sure if it works that way, or if I really want to take the time to do it.

**Comment by pjones112000** [comment]
I've run into this issue as well....I have a few old comics from Charlton Space: 1999 (1975) and Space: 1999 (1975).  Same publisher, same title, same release date...different issues.  I was able to edit the second one's folder changing it to Space 1999 Magazine (1975).  However, to make it work I had to mark all issues as wanted and then re-import them for both.  Both comics have different comicids.  I get it that my folder pattern doesn't include comicid otherwise it would have easily distinguished the difference.  I just wish it was a bit smarter somehow.
